src/service/viewport-impl.js

```diff
@@ -290,10 +296,11 @@ export class Viewport {
 
   /**
    * Returns the size of the viewport.
+   * @param {SizeReportingFidelity} fidelity
    * @return {!{width: number, height: number}}
    */
-  getSize() {
-    if (this.size_) {
+  getSize(fidelity = SizeReportingFidelity.LOW) {
+    if (this.size_ && fidelity == SizeReportingFidelity.LOW) {
```

Comment:
I've looked at the current master code. I see that the size is always refreshed on resize. Are you finding that that is not the case?